pandas/core/internals/blocks.py

```diff
@@ -1552,12 +1552,8 @@ def _slice(self, slicer):
     def getitem_block_index(self, slicer: slice) -> ExtensionBlock:
         """
         Perform __getitem__-like specialized to slicing along index.
-
-        Assumes self.ndim == 2
         """
-        # error: Invalid index type "Tuple[ellipsis, slice]" for
-        # "Union[ndarray, ExtensionArray]"; expected type "Union[int, slice, ndarray]"
-        new_values = self.values[..., slicer]  # type: ignore[index]
+        new_values = self.array_values[slicer]
```

Comment:
just .values, not .array_values

Reply:
can you add a comment to the effect of `# GH#42787 in principle equivalent to values[..., slicer] but we don't require subclasses to support that (for now)`

Reply:
Changed to `.values` and added an explanatory comment.